ros2topic/ros2topic/verb/pub.py

```diff
@@ -27,13 +34,85 @@
 from ros2topic.api import TopicNameCompleter
 from ros2topic.api import TopicTypeCompleter
 from ros2topic.verb import VerbExtension
-from rosidl_runtime_py import set_message_fields
+from rosidl_parser.definition import AbstractNestedType
+from rosidl_parser.definition import NamespacedType
+from rosidl_runtime_py.convert import get_message_slot_types
+from rosidl_runtime_py.import_message import import_message_from_namespaced_type
 from rosidl_runtime_py.utilities import get_message
 import yaml
 
 MsgType = TypeVar('MsgType')
 
 
+def set_message_fields_expanded(node: Node, msg: Any, values: Dict[str, str]) -> List[Any]:
```

Comment:
can you move most of the new logic to https://github.com/ros2/ros2cli/blob/0a33a5bd7fed869530b9dcf2e8771ff553d019dc/ros2topic/ros2topic/api/__init__.py?

Reply:
or maybe this could be added to `rosidl_runtime_py`, so it can be reused in other tools

Reply:
makes sense!

Reply:
@ivanpauno I prefer to move it to `rosidl_runtime_py` so that there's no duplicate logic. Moving to `ros2topic` may not be that useful since no other verbs use the original `set_message_fields` function